### ISSUE TITLE
#including a contract in another contract fails when they have a shared action name

### ISSUE THREAD

**Issue by swatanabe-b1** [EOSIO/eosio.cdt#543]
[include_contract_fail.tar.gz](https://github.com/EOSIO/eosio.cdt/files/3330544/include_contract_fail.tar.gz)

```
swatanabe:include_contract_fail steven.watanabe$ cleos push action accounting bad '[]' -p alice
Error 3050004: eosio_assert_code assertion failure
Error Details:
assertion failure with error code: 8000000000000000000
```

**Comment by jeffreyssmith2nd** [comment]
Confirmed on 1.6.2

**Comment by gravitt8460** [comment]
I also experience this. I also noticed that if two contracts share an on_notify indicator and they are in the same project (maybe one includes the other), I receive an error that the project already has an on_notify. However, they should be separate.  I work around this by creating multiple projects, but it does create code duplication by having to include headers in multiple places.

**Comment by SheldonHH** [comment]
+1

**Comment by swatanabe-b1** [comment]
It looks like this was fixed by a6477873c3f7b0b40af4eec8b1bc1afe51fd24b6

**Comment by swatanabe-b1** [comment]
#686
